mods/default/torch.lua

```diff
@@ -3,134 +3,104 @@
 -- support for MT game translation.
 local S = default.get_translator
 
-local function on_flood(pos, oldnode, newnode)
-	minetest.add_item(pos, ItemStack("default:torch 1"))
-	-- Play flame-extinguish sound if liquid is not an 'igniter'
-	local nodedef = minetest.registered_items[newnode.name]
-	if not (nodedef and nodedef.groups and
-			nodedef.groups.igniter and nodedef.groups.igniter > 0) then
-		minetest.sound_play(
-			"default_cool_lava",
-			{pos = pos, max_hear_distance = 16, gain = 0.1}
-		)
+function default.register_torch(name, def)
+	local torch = table.copy(def)
+	torch.drawtype = "mesh"
+	torch.paramtype = "light"
+	torch.paramtype2 = "wallmounted"
+	torch.sunlight_propagates = true
+	torch.walkable = false
+	torch.liquids_pointable = false
+	torch.drop = name
+	torch.floodable = true
+	torch.on_flood = function(pos, oldnode, newnode)
+		oldnode.name = name or name .. "_wall" or name .. "_celling"
```

Comment:
This will always be `oldnode.name = name` because `name` cannot be `nil`.
`minetest.add_item(pos, name)` will be sufficient here.

Reply:
I know. Therefore `name or name .. "_wall" or name .. "_celling"`

Reply:
```Lua
$ luajit
LuaJIT 2.1.0-beta3 -- Copyright (C) 2005-2017 Mike Pall. http://luajit.org/
JIT: ON SSE2 SSE3 fold cse dce fwd dse narrow loop abc sink fuse
> name = "torch"
> print(name or name .. "_wall" or name .. "_celling")
torch
> name = nil
> print(name or name .. "_wall" or name .. "_celling")
stdin:1: attempt to concatenate global 'name' (a nil value)
stack traceback:
	stdin:1: in main chunk
	[C]: at 0x56131ea0a1d0
```